// @rimiti/keldoc-js-sdk/dist/classes/exceptions/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BadRequest", {
  enumerable: true,
  get: function get() {
    return _bad_request.default;
  }
});
Object.defineProperty(exports, "InternalError", {
  enumerable: true,
  get: function get() {
    return _internal_error.default;
  }
});
Object.defineProperty(exports, "MissingMandatoryParameter", {
  enumerable: true,
  get: function get() {
    return _missing_mandatory_parameter.default;
  }
});
Object.defineProperty(exports, "NotAcceptable", {
  enumerable: true,
  get: function get() {
    return _not_acceptable.default;
  }
});
Object.defineProperty(exports, "NotFound", {
  enumerable: true,
  get: function get() {
    return _not_found.default;
  }
});
Object.defineProperty(exports, "OverQuota", {
  enumerable: true,
  get: function get() {
    return _over_quota.default;
  }
});
Object.defineProperty(exports, "Unauthorized", {
  enumerable: true,
  get: function get() {
    return _unauthorized.default;
  }
});
Object.defineProperty(exports, "ValidationError", {
  enumerable: true,
  get: function get() {
    return _validation_error.default;
  }
});
Object.defineProperty(exports, "InvalidDatetimeFormat", {
  enumerable: true,
  get: function get() {
    return _invalid_datetime_format.default;
  }
});
Object.defineProperty(exports, "UnknownError", {
  enumerable: true,
  get: function get() {
    return _unknown_error.default;
  }
});

var _bad_request = _interopRequireDefault(require("./bad_request"));

var _internal_error = _interopRequireDefault(require("./internal_error"));

var _missing_mandatory_parameter = _interopRequireDefault(require("./missing_mandatory_parameter"));

var _not_acceptable = _interopRequireDefault(require("./not_acceptable"));

var _not_found = _interopRequireDefault(require("./not_found"));

var _over_quota = _interopRequireDefault(require("./over_quota"));

var _unauthorized = _interopRequireDefault(require("./unauthorized"));

var _validation_error = _interopRequireDefault(require("./validation_error"));

var _invalid_datetime_format = _interopRequireDefault(require("./invalid_datetime_format"));

var _unknown_error = _interopRequireDefault(require("./unknown_error"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
